apps/prairielearn/src/lib/editors.js

```diff
@@ -1178,6 +1249,118 @@ export class FileUploadEditor extends Editor {
   }
 }
 
+export class FileModifyEditor extends Editor {
+  // Naming convention for contents and hashes in FileModifyEditor:
+  //
+  //    xyzContentsUTF - contents of xyz file as utf8
+  //
+  //    xyzContents - contents of xyz file as utf8 that is base64-encoded
+  //
+  //    xyzHash - hash of contents of xyz file as utf8 that is base64-encoded
+  //
+  // The base64 encoding and its corresponding hash are used by the file editor.
+  // If this weren't the case, then we wouldn't use it here either. For example,
+  // FileUploadEditor - which is used by the file browser - doesn't require any
+  // base64 encoding. In that case, contents/hashes are just utf8.
+
+  constructor(params) {
+    super(params);
+    this.container = params.container;
+    this.filePath = params.filePath;
+    this.editContents = params.editContents;
+    this.origHash = params.origHash;
+    if (this.course.path === this.container.rootPath) {
+      this.prefix = '';
+    } else {
+      this.prefix = `${path.basename(this.container.rootPath)}: `;
+    }
+    this.description = `${this.prefix}modify ${path.relative(
+      this.container.rootPath,
+      this.filePath,
+    )}`;
+  }
+
+  getHash(contents) {
+    return sha256(contents).toString();
+  }
+
+  shouldEdit(callback) {
+    debug('get hash of edit contents');
+    const editHash = this.getHash(this.editContents);
+    debug('editHash: ' + editHash);
+    debug('origHash: ' + this.origHash);
+    if (this.origHash === editHash) {
+      debug('edit contents are the same as orig contents, so abort');
+      callback(null, false);
+    } else {
+      debug('edit contents are different from orig contents, so continue');
+      callback(null, true);
+    }
+  }
```

Comment:
I'm not sure we need `shouldEdit`; is it really important to throw an error if the file contents are the same? I guess that could cause problems with `git` not finding any changed files to commit?